spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/handler/predicate/MethodRoutePredicateFactory.java

```diff
@@ -43,32 +43,39 @@ public MethodRoutePredicateFactory() {
 		return Arrays.asList(METHOD_KEY);
 	}
 
+	@Override
+	public ShortcutType shortcutType() {
+		return ShortcutType.GATHER_LIST;
+	}
+
 	@Override
 	public Predicate<ServerWebExchange> apply(Config config) {
 		return new GatewayPredicate() {
 			@Override
 			public boolean test(ServerWebExchange exchange) {
-				HttpMethod requestMethod = exchange.getRequest().getMethod();
-				return requestMethod == config.getMethod();
+				String requestMethod = exchange.getRequest().getMethod().name();
+				return config.getMethod().stream().anyMatch(
+						httpMethod -> requestMethod.equalsIgnoreCase(httpMethod));
 			}
 
 			@Override
 			public String toString() {
-				return String.format("Method: %s", config.getMethod());
+				return String.format("Methods: %s", config.getMethod());
 			}
 		};
 	}
 
+	@Validated
 	public static class Config {
 
-		private HttpMethod method;
+		private List<String> method;
```

Comment:
Why the change to string?

Reply:
My first implementation used List<HttpMethod> but then somehow `GatewayIntegrationTests::compositeRouteWorks` fails, because the generated Config just contains String values. I am still figuring out the root cause of this, therefore, I changed to String for now.

```
java.lang.ClassCastException: class java.lang.String cannot be cast to class org.springframework.http.HttpMethod (java.lang.String is in module java.base of loader 'bootstrap'; org.springframework.http.HttpMethod is in unnamed module of loader 'app')
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[na:na]
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1631) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[na:na]
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[na:na]
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528) ~[na:na]
	at org.springframework.cloud.gateway.handler.predicate.MethodRoutePredicateFactory$1.test(MethodRoutePredicateFactory.java:58) ~[classes/:na]
	at org.springframework.cloud.gateway.handler.predicate.MethodRoutePredicateFactory$1.test(MethodRoutePredicateFactory.java:54) ~[classes/:na]```

Reply:
As soon as I change to `ShortcutType.GATHER_LIST` it creates a List of Strings, `ShortcutType.DEFAULT` works but then obviously it only generates one value.

Have to look into it further.

Reply:
Changed to a good old array, works fine.

What do you think @spencergibb ?